server/src/main/model/Reserva.js

```diff
@@ -5,6 +5,17 @@
 
     const Schema = mongoose.Schema;
 
+    /**
+     * Dadas data e horario(inicio/fim) verifica se a data da reserva é válida.
+     * @param {String} data Dia da reserva no formato dd-MM-yyy
+     * @param {String} horario Horario da reserva no formato hh:mm
+     * @returns {Boolean} True caso a data seja válida
+     */
+    function isDataValida(data, horario){
```

Comment:
Pode colocar no util. Como `_.isDataFutura`. Inclusive, chamaria nesse arquivo com `_.isDataFutura(_.getData(this.dia, this.inicio))`